### PR TITLE
docs: new "how to debug slowness" article

### PR DESCRIPTION
**Pull request by aldeed**
Adds a new "How To: Debug slowness" article

### CODE REVIEW

**Comment by chrispotter** [comment]
`any layer or layer may be causing the slowness` did you mean `any layer or layers`?

File: public-docs/dev-debug-app-slowness.md [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+        $gt: ISODate("2019-04-26T02:29:39.315+0000")
+      }
+    }).sort({ ts: -1.0 });
+    ```
+
+## Step 5: Look at other potential causes
+
+1. If a query is slow but it used an index for lookup, often the slowness is caused by the `sort`. Does the slow query have a specified sort, and if so, is there an index that supports that sort? If not, create one locally and check whether the query is faster.
+    - Tip: A sort with multiple keys can only use an index with those same keys in the same order. The index direction need not be the same as the sort direction. For example, if the query sort is `{ createdAt: 1, _id: 1 }`, Mongo will use a `{ createdAt: 1, _id: 1 }` index or a `{ createdAt: -1, _id: -1 }` index, but it will not use a `{ createdAt: 1 }` index.
+2. If a query is optimized and MongoDB is using an index for both lookup and sorting, look at what fields are being used by the code. It's always a good idea to include the `projection` option with a list of only the fields you actually need. This may not make a huge difference in the speed, but it benefits memory and bandwidth as well.
```
Comment (spencern) [inline public-docs/dev-debug-app-slowness.md]: Glad you mentioned this. I think we’ve got a some queries where we’re returning a lot more fields that we need.

**Comment by aldeed** [comment]
@chrispotter Not sure what I meant but your suggestion makes sense. Will fix. 🤕

File: public-docs/dev-debug-app-slowness.md [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+Although this article won't delve into the subject, your first step when debugging slowness should be to rule out a network cause. Test on a fast network, with a wired connection, on a computer that is near the server's data center. If it's still slow, then proceed with the rest of this article.
+
+## Step 1: Identify and describe the causal action
+
+> Hint: Although the rest of this article is aimed at developers, anyone can do this step!
+
+Before you start reading code and examining database queries, be sure that you understand the issue in the most specific terms possible. Answer the following questions:
+
+- What user or system action begins the slowness? Document the specific reproduction steps.
+- How slow is it and how fast should it be to meet your users' expectations?
```
[the remark this answers is not in the document]
Comment (aldeed) [reply public-docs/dev-debug-app-slowness.md]: 👍 I'll add that

**Comment by aldeed** [comment]
Updated for corrections and additions from reviewers

**Pull request merged**